### ISSUE TITLE
Entity/Attribute Synchronization issues

### ISSUE THREAD

**Issue by torrin47** [Innovate-Inc/EMEMetadataToolKit#43]
The Synchronize Metadata GP tool seems to work ok for getting a list of attributes, but it doesn't obtain/synchronize the attribute properties - specifically Type and Width - which should be very easy to automatically derive. Seems like something we should try to get Esri to fix, and if they won't do it, we should do it ourselves. So like the other synchronization ticket #36, we should:

- [x] Confirm that behavior wasn't just user error
- [x] Submit ticket to Esri
- [ ] Explore alternate solutions

**Comment by torrin47** [comment]
It's possible that this issue was unique to me - a clean reinstall of ArcGIS Desktop recommended by Esri seems to have resolved the issue. Would someone mind testing entity-attribute synchronization on their local instances?

**Comment by torrin47** [comment]
Note to self... it appears that the "clean reinstall" broke EME.  Need to follow up and test to see whether EME is at fault for breaking standard Esri synchronizer.

**Comment by torrin47** [comment]
Ugh.  That's confirmed. The synchronizer works properly after a clean installation, but when EME 3.2.1 is installed, it blocks ArcCatalog from doing a full synchronization of all the ArcGIS/ISO metadata elements. It also doesn't appear to be resolved by simply uninstalling EME 3.2.1 - the changes caused by EME 3.2.1 seem to persist following the uninstall.  @dspinosa I know at one point you had the source code for EME 3.2.1 - I'm wondering whether that would provide any insight into what's being changed (probably a registry entry, right?) so that we could attempt to reverse it when we install our new tool - avoiding the need to uninstall EME 3.2.1 and potentially even do a full clean reinstall of ArcGIS Desktop?  Ugh.  Going to try a few more install/uninstall combos to diagnose, but this might end up being a new function we have to write.

**Comment by torrin47** [comment]
Ok, after a whole bunch of testing, I've traced the problem to a couple of Registry entries.  Here's a clean registry key:
```
[HKEY_CURRENT_USER\Software\ESRI\Desktop10.4\Metadata\Synchronizers]
"{1D8F7EFD-2B53-4045-BBDB-AB160C05A9F9}"=dword:00000001
```
and here's one with EME - there are two extra keys under Metadata\Synchronizers. 
```
[HKEY_CURRENT_USER\Software\ESRI\Desktop10.4\Metadata\Synchronizers]
"{1D8F7EFD-2B53-4045-BBDB-AB160C05A9F9}"=dword:00000001
"{671D8157-A8DC-49AB-B9F6-AAC239825110}"=dword:00000001
"{DA3DAD56-FABE-49F5-86AA-EFB146575A8E}"=dword:00000001
```
When one uninstalls EME 3.2.1, these entries are not removed - which is kind of sloppy.  Fortunately, after EME 3.2.1 is uninstalled, removing these keys manually resolves the issue without requiring a full reinstall of ArcGIS - which is awesome, because it means we could install a cleanup routine into the installation of the new EMEMetadataToolkit to resolve this issue.  The downside is that if you remove these entries with EME 3.2.1 installed, they just come back, which kinda means that we can't allow them to have EME 3.2.1 to hang around for legacy work, unless we have some other fix or patch to address the deeper synchronization problem. Anyway, nice to finally fully understand the problem even if we still have to discuss paths forward.
